In [11]:
import numpy as np
import pandas as pd 

In [12]:
#load the data

data = pd.read_excel('../data/Number_of_visits_to_UK.xlsx', engine="openpyxl")

In [13]:
data

Unnamed: 0  Unnamed: 1  Visits (thousands)  \
0                            NaN         NaN         2018.000000   
1                            NaN         NaN                 NaN   
2                         Canada         NaN          852.244006   
3                            USA         NaN         4571.476452   
4                  North America         NaN         5423.720459   
..                           ...         ...                 ...   
61                        Mexico         NaN          163.879608   
62  Other Central & Sth. America         NaN          302.287601   
63             Rest of the World         NaN            6.109402   
64               Other Countries         NaN         8166.856464   
65                   Total World         NaN        40283.202629   

      Unnamed: 3    Unnamed: 4   Unnamed: 5    Unnamed: 6  Unnamed: 7 2021-22  \
0    2019.000000          2020  2021.000000         2022R         NaN  (000s)   
1            NaN           NaN          NaN           NaN         NaN     NaN   
2     874.060640          -      128.184612    896.646943         NaN     768   
3    4498.754491          -      663.466616   4586.786268         NaN    3923   
4    5372.815130   1170.790477   791.651228   5483.433212         NaN    4692   
..           ...           ...          ...           ...         ...     ...   
61    166.197819          -        9.030770    163.132627         NaN     154   
62    310.557081          -       20.257408     227.35528         NaN     207   
63      1.963070          -        0.538417       4.43157         NaN       4   
64   8191.499487   1941.924295   758.567517   4780.865527         NaN    4022   
65  40857.447996  11100.507124  6383.823252  31244.438996         NaN   24861   

        2021-22.1  Unnamed: 11   2018-22  Unnamed: 13  Unnamed: 14  \
0             (%)          NaN       (%)          NaN          NaN   
1   NaN       NaN          NaN       NaN          NaN          NaN   
2           599.5          NaN  1.277828          NaN          NaN   
3           591.3          NaN   0.08362          NaN          NaN   
4           592.7          NaN   0.27411          NaN          NaN   
..  ...       ...          ...       ...          ...          ...   
61         1706.4          NaN -0.114148          NaN          NaN   
62         1022.3          NaN -6.873951          NaN          NaN   
63          723.1          NaN -7.713172          NaN          NaN   
64          530.2          NaN -12.52924          NaN          NaN   
65          389.4          NaN -6.154774          NaN          NaN   

    Unnamed: 15 Unnamed: 16  
0           NaN         NaN  
1           NaN         NaN  
2           NaN         NaN  
3           NaN         NaN  
4           NaN         NaN  
..          ...         ...  
61          NaN         NaN  
62          NaN         NaN  
63          NaN         NaN  
64          NaN         NaN  
65          NaN         NaN  

[66 rows x 17 columns]

In [14]:
data1 = data.iloc[:,:7]
data1

Unnamed: 0  Unnamed: 1  Visits (thousands)  \
0                            NaN         NaN         2018.000000   
1                            NaN         NaN                 NaN   
2                         Canada         NaN          852.244006   
3                            USA         NaN         4571.476452   
4                  North America         NaN         5423.720459   
..                           ...         ...                 ...   
61                        Mexico         NaN          163.879608   
62  Other Central & Sth. America         NaN          302.287601   
63             Rest of the World         NaN            6.109402   
64               Other Countries         NaN         8166.856464   
65                   Total World         NaN        40283.202629   

      Unnamed: 3    Unnamed: 4   Unnamed: 5    Unnamed: 6  
0    2019.000000          2020  2021.000000         2022R  
1            NaN           NaN          NaN           NaN  
2     874.060640          -      128.184612    896.646943  
3    4498.754491          -      663.466616   4586.786268  
4    5372.815130   1170.790477   791.651228   5483.433212  
..           ...           ...          ...           ...  
61    166.197819          -        9.030770    163.132627  
62    310.557081          -       20.257408     227.35528  
63      1.963070          -        0.538417       4.43157  
64   8191.499487   1941.924295   758.567517   4780.865527  
65  40857.447996  11100.507124  6383.823252  31244.438996  

[66 rows x 7 columns]

In [15]:
#change columns 
data2 = data1.copy()

new_column_names = {'Unnamed: 0': 'country','Visits (thousands)': '2018', 'Unnamed: 3': '2019', 'Unnamed: 4': '2020', 'Unnamed: 5': '2021','Unnamed: 6': '2022'}
data2 = data2.rename(columns=new_column_names)
data2

country  Unnamed: 1          2018          2019  \
0                            NaN         NaN   2018.000000   2019.000000   
1                            NaN         NaN           NaN           NaN   
2                         Canada         NaN    852.244006    874.060640   
3                            USA         NaN   4571.476452   4498.754491   
4                  North America         NaN   5423.720459   5372.815130   
..                           ...         ...           ...           ...   
61                        Mexico         NaN    163.879608    166.197819   
62  Other Central & Sth. America         NaN    302.287601    310.557081   
63             Rest of the World         NaN      6.109402      1.963070   
64               Other Countries         NaN   8166.856464   8191.499487   
65                   Total World         NaN  40283.202629  40857.447996   

            2020         2021          2022  
0           2020  2021.000000         2022R  
1            NaN          NaN           NaN  
2           -      128.184612    896.646943  
3           -      663.466616   4586.786268  
4    1170.790477   791.651228   5483.433212  
..           ...          ...           ...  
61          -        9.030770    163.132627  
62          -       20.257408     227.35528  
63          -        0.538417       4.43157  
64   1941.924295   758.567517   4780.865527  
65  11100.507124  6383.823252  31244.438996  

[66 rows x 7 columns]

In [16]:
#remove 'unnamed: 1'
data3 = data2.drop('Unnamed: 1', axis =1)
data3

country          2018          2019          2020  \
0                            NaN   2018.000000   2019.000000          2020   
1                            NaN           NaN           NaN           NaN   
2                         Canada    852.244006    874.060640          -      
3                            USA   4571.476452   4498.754491          -      
4                  North America   5423.720459   5372.815130   1170.790477   
..                           ...           ...           ...           ...   
61                        Mexico    163.879608    166.197819          -      
62  Other Central & Sth. America    302.287601    310.557081          -      
63             Rest of the World      6.109402      1.963070          -      
64               Other Countries   8166.856464   8191.499487   1941.924295   
65                   Total World  40283.202629  40857.447996  11100.507124   

           2021          2022  
0   2021.000000         2022R  
1           NaN           NaN  
2    128.184612    896.646943  
3    663.466616   4586.786268  
4    791.651228   5483.433212  
..          ...           ...  
61     9.030770    163.132627  
62    20.257408     227.35528  
63     0.538417       4.43157  
64   758.567517   4780.865527  
65  6383.823252  31244.438996  

[66 rows x 6 columns]

In [17]:
#remove index 0 and 1 
data4 = data3.iloc[2:,:]
data4

country          2018          2019          2020  \
2                         Canada    852.244006    874.060640          -      
3                            USA   4571.476452   4498.754491          -      
4                  North America   5423.720459   5372.815130   1170.790477   
5                        Austria    327.289461    342.130139          -      
6                        Belgium   1096.778896   1134.973111          -      
..                           ...           ...           ...           ...   
61                        Mexico    163.879608    166.197819          -      
62  Other Central & Sth. America    302.287601    310.557081          -      
63             Rest of the World      6.109402      1.963070          -      
64               Other Countries   8166.856464   8191.499487   1941.924295   
65                   Total World  40283.202629  40857.447996  11100.507124   

           2021          2022  
2    128.184612    896.646943  
3    663.466616   4586.786268  
4    791.651228   5483.433212  
5     55.705091    288.058999  
6    159.356974    679.136413  
..          ...           ...  
61     9.030770    163.132627  
62    20.257408     227.35528  
63     0.538417       4.43157  
64   758.567517   4780.865527  
65  6383.823252  31244.438996  

[64 rows x 6 columns]

In [ ]:
# %load ../src/function.py
import pandas as pd

def remove_region(df:pd.DataFrame) -> pd.DataFrame:
	'''
	This function removes enexpected values(region) of the column: 'country' but only' total world' and 
	something of 'other--- ' are remained.

	Input:pd.DataFrame

	Output:pd.DataFrame  


	''' 
	df2 = df.copy()

	remove_regions = {'North America', 'North America', '- of which EU', '- of which EU15', '- of which EU Oth', 'Europe'}

	#for i in len(df):
	#	if df2['country'] in remove_regions:
	#		df2. 
	df2 = df2[~df2['country'].isin(remove_regions )]





	return df2




#if __name__ == "__main__" :
	
#	 remove_region()










In [19]:
%run -i ../src/function.py

data5 = data4.copy()
data6 = remove_region(data5)
data6

country          2018          2019          2020  \
2                         Canada    852.244006    874.060640          -      
3                            USA   4571.476452   4498.754491          -      
5                        Austria    327.289461    342.130139          -      
6                        Belgium   1096.778896   1134.973111          -      
7                       Bulgaria    232.804800    235.485153          -      
8                 Czech Republic    360.292490    413.526195          -      
9                         Cyprus    108.084657    128.453580          -      
10                       Denmark    614.176524    691.394210          -      
11                       Finland    188.279552    214.525742          -      
12                        France   3591.885263   3560.516774          -      
13                       Germany   3170.208595   3232.912268          -      
14                        Greece    223.120404    250.185039          -      
15                       Hungary    371.741556    333.702589          -      
16           Republic of Ireland   2982.710728   2851.180167          -      
17                         Italy   2080.462873   2196.629409          -      
18                     Lithuania    321.866337    235.956287          -      
19                    Luxembourg     78.018108    145.859117          -      
20                         Malta    111.271345    189.763566          -      
21                   Netherlands   1978.292201   1986.770528          -      
22                        Norway    647.814092    647.460914          -      
23                        Poland   1555.325884   1651.433620          -      
24                      Portugal    452.922483    612.689278          -      
25                       Romania    896.580832    902.296778          -      
26                        Russia    228.772140    198.991748          -      
27                      Slovakia    119.708697    112.570385          -      
28                         Spain   2450.872835   2318.721901          -      
29                        Sweden    745.030397    789.407605          -      
30                   Switzerland    825.256645    925.727177          -      
31                        Turkey    317.599597    334.448240          -      
32                Rest of Europe    615.458314    655.421860          -      
37                         Egypt     56.186585     52.027322          -      
38                       Morocco     30.714891     33.358548          -      
39                       Tunisia      4.470442      8.978016          -      
40            Other North Africa     22.240064     20.991022          -      
41                  South Africa    264.985586    217.116994          -      
42                       Nigeria    203.225821    208.367278          -      
43                  Other Africa    203.969192    187.937138          -      
44                        Israel    263.752452    264.664375          -      
45          United Arab Emirates    511.465054    552.898500          -      
46             Other Middle East    651.384884    723.432646          -      
47             Hong Kong (China)    402.721910    408.963095          -      
48                   Other China   1001.447851   1010.060521          -      
49                         India    679.585386    692.082857          -      
50                         Japan    428.952132    388.839266          -      
51                      Pakistan     94.928397     89.674391          -      
52                     Sri Lanka     17.936838     28.525133          -      
53                      Thailand    188.977827    184.416425          -      
54                    Other Asia    976.096696    954.729938          -      
55                     Australia   1039.029263   1063.404082          -      
56                   New Zealand    231.717674    187.682465          -      
57                      Barbados     23.328468 

In [21]:
# replace '-' into null values 
data6.dtypes

country     object
2018       float64
2019       float64
2020        object
2021       float64
2022        object
dtype: object

In [22]:
# the column '2020' is changed into null values and float values. 
# first change the index 
data8 = data6.reset_index(drop=True)
data8

country          2018          2019          2020  \
0                         Canada    852.244006    874.060640          -      
1                            USA   4571.476452   4498.754491          -      
2                        Austria    327.289461    342.130139          -      
3                        Belgium   1096.778896   1134.973111          -      
4                       Bulgaria    232.804800    235.485153          -      
5                 Czech Republic    360.292490    413.526195          -      
6                         Cyprus    108.084657    128.453580          -      
7                        Denmark    614.176524    691.394210          -      
8                        Finland    188.279552    214.525742          -      
9                         France   3591.885263   3560.516774          -      
10                       Germany   3170.208595   3232.912268          -      
11                        Greece    223.120404    250.185039          -      
12                       Hungary    371.741556    333.702589          -      
13           Republic of Ireland   2982.710728   2851.180167          -      
14                         Italy   2080.462873   2196.629409          -      
15                     Lithuania    321.866337    235.956287          -      
16                    Luxembourg     78.018108    145.859117          -      
17                         Malta    111.271345    189.763566          -      
18                   Netherlands   1978.292201   1986.770528          -      
19                        Norway    647.814092    647.460914          -      
20                        Poland   1555.325884   1651.433620          -      
21                      Portugal    452.922483    612.689278          -      
22                       Romania    896.580832    902.296778          -      
23                        Russia    228.772140    198.991748          -      
24                      Slovakia    119.708697    112.570385          -      
25                         Spain   2450.872835   2318.721901          -      
26                        Sweden    745.030397    789.407605          -      
27                   Switzerland    825.256645    925.727177          -      
28                        Turkey    317.599597    334.448240          -      
29                Rest of Europe    615.458314    655.421860          -      
30                         Egypt     56.186585     52.027322          -      
31                       Morocco     30.714891     33.358548          -      
32                       Tunisia      4.470442      8.978016          -      
33            Other North Africa     22.240064     20.991022          -      
34                  South Africa    264.985586    217.116994          -      
35                       Nigeria    203.225821    208.367278          -      
36                  Other Africa    203.969192    187.937138          -      
37                        Israel    263.752452    264.664375          -      
38          United Arab Emirates    511.465054    552.898500          -      
39             Other Middle East    651.384884    723.432646          -      
40             Hong Kong (China)    402.721910    408.963095          -      
41                   Other China   1001.447851   1010.060521          -      
42                         India    679.585386    692.082857          -      
43                         Japan    428.952132    388.839266          -      
44                      Pakistan     94.928397     89.674391          -      
45                     Sri Lanka     17.936838     28.525133          -      
46                      Thailand    188.977827    184.416425          -      
47                    Other Asia    976.096696    954.729938          -      
48                     Australia   1039.029263   1063.404082          -      
49                   New Zealand    231.717674    187.682465          -      
50                      Barbados     23.328468 

In [23]:
data8.loc[0:56,'2020'] = np.nan
data8

country          2018          2019          2020  \
0                         Canada    852.244006    874.060640           NaN   
1                            USA   4571.476452   4498.754491           NaN   
2                        Austria    327.289461    342.130139           NaN   
3                        Belgium   1096.778896   1134.973111           NaN   
4                       Bulgaria    232.804800    235.485153           NaN   
5                 Czech Republic    360.292490    413.526195           NaN   
6                         Cyprus    108.084657    128.453580           NaN   
7                        Denmark    614.176524    691.394210           NaN   
8                        Finland    188.279552    214.525742           NaN   
9                         France   3591.885263   3560.516774           NaN   
10                       Germany   3170.208595   3232.912268           NaN   
11                        Greece    223.120404    250.185039           NaN   
12                       Hungary    371.741556    333.702589           NaN   
13           Republic of Ireland   2982.710728   2851.180167           NaN   
14                         Italy   2080.462873   2196.629409           NaN   
15                     Lithuania    321.866337    235.956287           NaN   
16                    Luxembourg     78.018108    145.859117           NaN   
17                         Malta    111.271345    189.763566           NaN   
18                   Netherlands   1978.292201   1986.770528           NaN   
19                        Norway    647.814092    647.460914           NaN   
20                        Poland   1555.325884   1651.433620           NaN   
21                      Portugal    452.922483    612.689278           NaN   
22                       Romania    896.580832    902.296778           NaN   
23                        Russia    228.772140    198.991748           NaN   
24                      Slovakia    119.708697    112.570385           NaN   
25                         Spain   2450.872835   2318.721901           NaN   
26                        Sweden    745.030397    789.407605           NaN   
27                   Switzerland    825.256645    925.727177           NaN   
28                        Turkey    317.599597    334.448240           NaN   
29                Rest of Europe    615.458314    655.421860           NaN   
30                         Egypt     56.186585     52.027322           NaN   
31                       Morocco     30.714891     33.358548           NaN   
32                       Tunisia      4.470442      8.978016           NaN   
33            Other North Africa     22.240064     20.991022           NaN   
34                  South Africa    264.985586    217.116994           NaN   
35                       Nigeria    203.225821    208.367278           NaN   
36                  Other Africa    203.969192    187.937138           NaN   
37                        Israel    263.752452    264.664375           NaN   
38          United Arab Emirates    511.465054    552.898500           NaN   
39             Other Middle East    651.384884    723.432646           NaN   
40             Hong Kong (China)    402.721910    408.963095           NaN   
41                   Other China   1001.447851   1010.060521           NaN   
42                         India    679.585386    692.082857           NaN   
43                         Japan    428.952132    388.839266           NaN   
44                      Pakistan     94.928397     89.674391           NaN   
45                     Sri Lanka     17.936838     28.525133           NaN   
46                      Thailand    188.977827    184.416425           NaN   
47                    Other Asia    976.096696    954.729938           NaN   
48                     Australia   1039.029263   1063.404082           NaN   
49                   New Zealand    231.717674    187.682465           NaN   
50                      Barbados     23.328468 

In [25]:
# change the data type of '2022' into float64
data8['2022'] = data8['2022'].astype('float64')
data8.dtypes

country     object
2018       float64
2019       float64
2020       float64
2021       float64
2022       float64
dtype: object

In the process of importing the Excel file, all values were unexpectedly changed. For example, the value of Canada (2018) is 852, but it says 852.244006. So, it has to be rounded.
In addition, the unit is 'thousands. As the small amount of number is better to look at, the value will not be multiplied by 1000. 

In [27]:
data9 = data8.copy()
data10 = data9.round()
data10

country     2018     2019     2020    2021     2022
0                         Canada    852.0    874.0      NaN   128.0    897.0
1                            USA   4571.0   4499.0      NaN   663.0   4587.0
2                        Austria    327.0    342.0      NaN    56.0    288.0
3                        Belgium   1097.0   1135.0      NaN   159.0    679.0
4                       Bulgaria    233.0    235.0      NaN    56.0    183.0
5                 Czech Republic    360.0    414.0      NaN    55.0    276.0
6                         Cyprus    108.0    128.0      NaN    42.0    123.0
7                        Denmark    614.0    691.0      NaN   102.0    566.0
8                        Finland    188.0    215.0      NaN    17.0    169.0
9                         France   3592.0   3561.0      NaN   677.0   2830.0
10                       Germany   3170.0   3233.0      NaN   390.0   2233.0
11                        Greece    223.0    250.0      NaN    72.0    185.0
12                       Hungary    372.0    334.0      NaN    58.0    255.0
13           Republic of Ireland   2983.0   2851.0      NaN   708.0   2509.0
14                         Italy   2080.0   2197.0      NaN   256.0   1313.0
15                     Lithuania    322.0    236.0      NaN    47.0    164.0
16                    Luxembourg     78.0    146.0      NaN    17.0     82.0
17                         Malta    111.0    190.0      NaN    28.0     70.0
18                   Netherlands   1978.0   1987.0      NaN   273.0   1632.0
19                        Norway    648.0    647.0      NaN    75.0    547.0
20                        Poland   1555.0   1651.0      NaN   337.0   1345.0
21                      Portugal    453.0    613.0      NaN   147.0    494.0
22                       Romania    897.0    902.0      NaN   191.0    634.0
23                        Russia    229.0    199.0      NaN    23.0     44.0
24                      Slovakia    120.0    113.0      NaN    12.0     77.0
25                         Spain   2451.0   2319.0      NaN   525.0   2012.0
26                        Sweden    745.0    789.0      NaN   117.0    630.0
27                   Switzerland    825.0    926.0      NaN   164.0    862.0
28                        Turkey    318.0    334.0      NaN    30.0    187.0
29                Rest of Europe    615.0    655.0      NaN   201.0    592.0
30                         Egypt     56.0     52.0      NaN     8.0     31.0
31                       Morocco     31.0     33.0      NaN     2.0     15.0
32                       Tunisia      4.0      9.0      NaN     1.0      9.0
33            Other North Africa     22.0     21.0      NaN     1.0     13.0
34                  South Africa    265.0    217.0      NaN    17.0    207.0
35                       Nigeria    203.0    208.0      NaN    80.0    207.0
36                  Other Africa    204.0    188.0      NaN    27.0    127.0
37                        Israel    264.0    265.0      NaN    71.0    245.0
38          United Arab Emirates    511.0    553.0      NaN    91.0    380.0
39             Other Middle East    651.0    723.0      NaN   158.0    470.0
40             Hong Kong (China)    403.0    409.0      NaN    12.0    112.0
41                   Other China   1001.0   1010.0      NaN    17.0     73.0
42                         India    680.0    692.0      NaN    58.0    516.0
43                         Japan    429.0    389.0      NaN    10.0    115.0
44                      Pakistan     95.0     90.0      NaN    20.0     85.0
45                     Sri Lanka     18.0     29.0      NaN     1.0     13.0
46                      Thailand    189.0    184.0      NaN     5.0     82.0
47                    Other Asia    976.0    955.0      NaN    75.0    484.0
48                     Australia   1039.0   1063.0      NaN    24.0    723.0
49                   New Zealand    232.0    188.0      NaN     9.0    126.0
50                      Barbados     23.0     27.0      NaN     2.0      8.0
51                   

In [33]:
# The data type apart from the column'country' changed into 'int64'.
data11 = data10.copy()
data12 = data11.astype('int64', errors='ignore')
data12

country   2018   2019     2020  2021   2022
0                         Canada    852    874      NaN   128    897
1                            USA   4571   4499      NaN   663   4587
2                        Austria    327    342      NaN    56    288
3                        Belgium   1097   1135      NaN   159    679
4                       Bulgaria    233    235      NaN    56    183
5                 Czech Republic    360    414      NaN    55    276
6                         Cyprus    108    128      NaN    42    123
7                        Denmark    614    691      NaN   102    566
8                        Finland    188    215      NaN    17    169
9                         France   3592   3561      NaN   677   2830
10                       Germany   3170   3233      NaN   390   2233
11                        Greece    223    250      NaN    72    185
12                       Hungary    372    334      NaN    58    255
13           Republic of Ireland   2983   2851      NaN   708   2509
14                         Italy   2080   2197      NaN   256   1313
15                     Lithuania    322    236      NaN    47    164
16                    Luxembourg     78    146      NaN    17     82
17                         Malta    111    190      NaN    28     70
18                   Netherlands   1978   1987      NaN   273   1632
19                        Norway    648    647      NaN    75    547
20                        Poland   1555   1651      NaN   337   1345
21                      Portugal    453    613      NaN   147    494
22                       Romania    897    902      NaN   191    634
23                        Russia    229    199      NaN    23     44
24                      Slovakia    120    113      NaN    12     77
25                         Spain   2451   2319      NaN   525   2012
26                        Sweden    745    789      NaN   117    630
27                   Switzerland    825    926      NaN   164    862
28                        Turkey    318    334      NaN    30    187
29                Rest of Europe    615    655      NaN   201    592
30                         Egypt     56     52      NaN     8     31
31                       Morocco     31     33      NaN     2     15
32                       Tunisia      4      9      NaN     1      9
33            Other North Africa     22     21      NaN     1     13
34                  South Africa    265    217      NaN    17    207
35                       Nigeria    203    208      NaN    80    207
36                  Other Africa    204    188      NaN    27    127
37                        Israel    264    265      NaN    71    245
38          United Arab Emirates    511    553      NaN    91    380
39             Other Middle East    651    723      NaN   158    470
40             Hong Kong (China)    403    409      NaN    12    112
41                   Other China   1001   1010      NaN    17     73
42                         India    680    692      NaN    58    516
43                         Japan    429    389      NaN    10    115
44                      Pakistan     95     90      NaN    20     85
45                     Sri Lanka     18     29      NaN     1     13
46                      Thailand    189    184      NaN     5     82
47                    Other Asia    976    955      NaN    75    484
48                     Australia   1039   1063      NaN    24    723
49                   New Zealand    232    188      NaN     9    126
50                      Barbados     23     27      NaN     2      8
51                       Jamaica     15      8      NaN     2      6
52               Other Caribbean     95    109      NaN    18     74
53                        Brazil    288    291      NaN    19    267
54                        Mexico    164    166      NaN     9    163
55  Other Central & Sth. America    302    311      NaN    20    227
56             Rest of the World      6      2      NaN     1      4
57               Other 

The values of '2020' have null values, so the int values are converted into float values. 

In [47]:
data12

country   2018   2019     2020  2021   2022
0                         Canada    852    874      NaN   128    897
1                            USA   4571   4499      NaN   663   4587
2                        Austria    327    342      NaN    56    288
3                        Belgium   1097   1135      NaN   159    679
4                       Bulgaria    233    235      NaN    56    183
5                 Czech Republic    360    414      NaN    55    276
6                         Cyprus    108    128      NaN    42    123
7                        Denmark    614    691      NaN   102    566
8                        Finland    188    215      NaN    17    169
9                         France   3592   3561      NaN   677   2830
10                       Germany   3170   3233      NaN   390   2233
11                        Greece    223    250      NaN    72    185
12                       Hungary    372    334      NaN    58    255
13           Republic of Ireland   2983   2851      NaN   708   2509
14                         Italy   2080   2197      NaN   256   1313
15                     Lithuania    322    236      NaN    47    164
16                    Luxembourg     78    146      NaN    17     82
17                         Malta    111    190      NaN    28     70
18                   Netherlands   1978   1987      NaN   273   1632
19                        Norway    648    647      NaN    75    547
20                        Poland   1555   1651      NaN   337   1345
21                      Portugal    453    613      NaN   147    494
22                       Romania    897    902      NaN   191    634
23                        Russia    229    199      NaN    23     44
24                      Slovakia    120    113      NaN    12     77
25                         Spain   2451   2319      NaN   525   2012
26                        Sweden    745    789      NaN   117    630
27                   Switzerland    825    926      NaN   164    862
28                        Turkey    318    334      NaN    30    187
29                Rest of Europe    615    655      NaN   201    592
30                         Egypt     56     52      NaN     8     31
31                       Morocco     31     33      NaN     2     15
32                       Tunisia      4      9      NaN     1      9
33            Other North Africa     22     21      NaN     1     13
34                  South Africa    265    217      NaN    17    207
35                       Nigeria    203    208      NaN    80    207
36                  Other Africa    204    188      NaN    27    127
37                        Israel    264    265      NaN    71    245
38          United Arab Emirates    511    553      NaN    91    380
39             Other Middle East    651    723      NaN   158    470
40             Hong Kong (China)    403    409      NaN    12    112
41                   Other China   1001   1010      NaN    17     73
42                         India    680    692      NaN    58    516
43                         Japan    429    389      NaN    10    115
44                      Pakistan     95     90      NaN    20     85
45                     Sri Lanka     18     29      NaN     1     13
46                      Thailand    189    184      NaN     5     82
47                    Other Asia    976    955      NaN    75    484
48                     Australia   1039   1063      NaN    24    723
49                   New Zealand    232    188      NaN     9    126
50                      Barbados     23     27      NaN     2      8
51                       Jamaica     15      8      NaN     2      6
52               Other Caribbean     95    109      NaN    18     74
53                        Brazil    288    291      NaN    19    267
54                        Mexico    164    166      NaN     9    163
55  Other Central & Sth. America    302    311      NaN    20    227
56             Rest of the World      6      2      NaN     1      4
57               Other 

In [48]:
# save the data
data12.to_csv('../data/cleaned_number_of_visitors.csv', index=False)
